libcloud/storage/drivers/s3.py

```diff
@@ -1015,8 +1015,35 @@ class S3SAEastStorageDriver(S3StorageDriver):
     ex_location_name = 'sa-east-1'
 
 
-class S3RGWOutscaleConnection(S3Connection):
-    pass
+def _get_s3_rgw_outscale_conn_by_sign_version(signature_version):
+
+    # aws4 connection support
+    class S3RGWOutscaleConnectionAWS4(SignedAWSConnection, BaseS3Connection):
```

Comment:
I would prefer to make those classes top-level and define them outside of this function, but that's mostly a style thing.